src/Symfony/Component/Console/Application.php

```diff
@@ -769,8 +769,8 @@ protected function configureIO(InputInterface $input, OutputInterface $output)
 
         if (true === $input->hasParameterOption(array('--no-interaction', '-n'), true)) {
             $input->setInteractive(false);
-        } elseif (function_exists('posix_isatty') && $this->getHelperSet()->has('question')) {
-            $inputStream = $this->getHelperSet()->get('question')->getInputStream();
```

Comment:
Good catch. 
Detecting a possible null helper then null inputStream here would be much logic.
I would prefer calling `Input::setStream` in `QuestionHelper::setInputStream` while setting the `QuestionHelper::$inputStream` (for BC reasons too) then return it in `QuestionHelper::getInputStream()` assuming that if it is set in the QuestionHelper, it will not be from the Input in the meantime.